src/features/contracts/contracts-directory.ts

```diff
@@ -8,13 +8,82 @@ import * as stream from 'stream';
 import * as path from 'path';
 import * as os from 'os';
 import type { RepositoryInfo, Contract } from './index';
+import { isValidUrl } from '../../lib/utils';
 
 const pipeline = util.promisify(stream.pipeline);
 const exists = util.promisify(fs.exists);
 const mkdir = util.promisify(fs.mkdir);
 
 const CONTRACTS_BASE_DIR = path.join(os.tmpdir(), 'contracts');
 
+const getDataUri = async (
+	filePath: string,
+	mimeType: 'image/png' | 'image/svg+xml',
+) => {
+	const base64Content = (await fs.promises.readFile(filePath)).toString(
+		'base64',
+	);
+	return `data:${mimeType};base64,${base64Content}`;
+};
+
+// All assets that are stored together with the contract are encoded and stored in a dataurl format.
+const handleLocalAssetUrl = async (assetUrl: string): Promise<string> => {
+	switch (path.extname(assetUrl)) {
+		case '.svg': {
+			return getDataUri(assetUrl, 'image/svg+xml');
+		}
+		case '.png': {
+			return getDataUri(assetUrl, 'image/png');
+		}
+		default:
+			return assetUrl;
```

Comment:
Would it make more sense to just return `null` instead of the fs path?

Reply:
yeah maybe, lets do that

Reply:
The Contract spec requires that there is a URL, if you provided an asset, so we need to return a string. It can be an empty one though, let's do that.